staticjinja/dep_graph.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding:utf-8 -*-
+
+"""
+Dependency graph for staticjinja
+"""
+
+
+class DepGraph(object):
+    """
+    A directed graph which will handle dependencies between sources in a site.
+
+    :param parents:
+        A dictionary whose keys are vertices of the graph and each value is the
+        set of child vertices of the key vertex.
+
+    :param children:
+        A dictionary whose keys are vertices of the graph and each value is the
+        set of parents vertices of the key vertex.
+
+    """
+    def __init__(self, parents, children):
+        self.parents = parents
+        self.children = children
+
+    @classmethod
+    def from_parents(cls, parents):
+        """
+        Buils a graph from a dictionnary of directed edges
+        by building the dictionnary of children edges.
```

Comment:
I'm sorry about that. I don't why I enable spell check while typing LaTeX and not python, maybe I fear it will highlight python stuff. I'll also try to hunt down those if we move forward in this direction.